src/app/qgssettingstree.cpp

```diff
@@ -60,37 +60,35 @@ QgsSettingsTree::QgsSettingsTree( QWidget *parent )
   header()->resizeSection( 1, 100 );
   header()->resizeSection( 2, 250 );
 
-  settings = nullptr;
-  refreshTimer.setInterval( 2000 );
-  autoRefresh = false;
+  mRefreshTimer.setInterval( 2000 );
 
-  groupIcon.addPixmap( style()->standardPixmap( QStyle::SP_DirClosedIcon ),
-                       QIcon::Normal, QIcon::Off );
-  groupIcon.addPixmap( style()->standardPixmap( QStyle::SP_DirOpenIcon ),
-                       QIcon::Normal, QIcon::On );
-  keyIcon.addPixmap( style()->standardPixmap( QStyle::SP_FileIcon ) );
+  mGroupIcon.addPixmap( style()->standardPixmap( QStyle::SP_DirClosedIcon ),
+                        QIcon::Normal, QIcon::Off );
+  mGroupIcon.addPixmap( style()->standardPixmap( QStyle::SP_DirOpenIcon ),
+                        QIcon::Normal, QIcon::On );
+  mKeyIcon = style()->standardIcon( QStyle::SP_FileIcon );
 
   setEditTriggers( QAbstractItemView::AllEditTriggers );
 
-  connect( &refreshTimer, &QTimer::timeout, this, &QgsSettingsTree::maybeRefresh );
+  connect( &mRefreshTimer, &QTimer::timeout, this, &QgsSettingsTree::maybeRefresh );
 }
 
 void QgsSettingsTree::setSettingsObject( QgsSettings *settings )
 {
-  delete this->settings;
-  this->settings = settings;
+  delete this->mSettings;
```

Comment:
How is the ownership of `mSettings` managed? It's deleted here, but not in the destructor and the signature also doesn't speak about transferring ownership.

Reply:
Note sure. Ping @3nids

Reply:
Generally speaking we shouldn't do this but use `QSettings()...` directly any time we need it, according to Qt docs this is fast enough ( http://doc.qt.io/qt-5/qsettings.html#basic-usage second paragraph )